sips/sip-16/sip-016-token-metadata.md

```diff
@@ -0,0 +1,394 @@
+        },
+        "animation_url": {
+            "type": "string",
+            "description": "URL to a multi-media attachment for the item. Deprecated. Use `properties.animation_url`."
+        },
+    }
+}
+```
+
+The lengths of string values is not restricted. Nowadays, clients should be smart enough to deal with values of different lengths.
+
+### Example
+
+token101.json
+
+```
+{
+  "sip": 16,
+  "name": "Foo #101",
+  "image": "ipfs://somerandomecid",
+  "attributes": [
+     {
+      "trait_type": "hair",
+      "value": "red",
+    },
+    {
+      "trait_type": "strength",
+      "display_type": "number",
+      "value": 99,
+    },
+  ],
+  "properties": {
+      "collection":  "Foo Collection",
+      "total_supply":  "10000"
+  },
+  "localization": {
+      "uri": "ipfs://somerandomcid/{locale}.json",
+      "default": "en",
+      "locales": ["en", "pt-BR", "de"]
+  }
+}
+```
+
+de.json
+
+```
+{
+    "sip": 16,
+    "attributes: [
+        {
+          "trait_type": "Haare",
+          "value": "rot",
+        },
+        {
+          "trait_type": "Stärke",
+          "display_type": "number",
+          "value": 99,
+        },
+    ]
+}
+```
+
+pt-BR.json
+
+```
+{
+    "sip": 16,
+    "attributes: [
+        {
+          "trait_type": "cabelos",
+          "value": "vermelho",
+        },
+        {
+          "trait_type": "amido",
+          "display_type": "number",
+          "value": 99,
+        },
+    ]
+}
+```
+
+### Properties
+
+Common properties are described in appendix C. Properties of type object are usually described using the following schema:
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata Property",
+    "type": "object",
+    "required": [],
+    "properties": {
+        "type": {
+            "type": "string",
+            "description": "type of the property"
+        },
+        "description": {
+            "type": "string",
+            "description": "description of the property"
+        },
+        "value": {
+            "type": {"oneOf": [{"type": "object"}, {"type": "string"}, {"type": "number"}, {"type": "integer"}, {"type": "boolean"} {"type: "array"}},
+            "description": "value of the property"
+        }
+    }
+}
+```
+
+Example:
+
+```
+{
+    "type": "string",
+    "description": "Address of custodian key holder",
+    "value": "Casa Inc., P.O. Box 20575, Charleston, S.C. 29413, United States."
+}
+```
+
+### Attributes
+
+Attributes describe addition elements of tokens that are "observable", usually represented in the image of the token.
+
+In contrast, properties describe elements of tokens that are more abstract.
+
+An attribute consists of a `trait_type` defining the name of the trait, `value` is the value of the trait, and `display_type` is a field indicating how you would like it to be displayed.
+
+Appendix B describes type of attributes
```

Comment:
After reading and re-reading this section, I still don't really know the difference between `attributes` and `properties`.  Can you provide a deeper explanation?  I also still don't really understand what `trait_type` and `display_type` do, and why they're necessary.

Reply:
I see that.  Thanks!